File: 9_special_pythagorean_triplet/9_special_pythagorean_triplet.py
# Problem is here: https://projecteuler.net/problem=9
# finding the product abc of the pythagorean triplet a^2 + b^2 = c^2 where a + b + c = 1000

def main():
    for a in range(1,333):
        for b in range(a,500):
            for c in range(b,1000):
                if a*a + b*b == c*c:
                    if a+b+c == 1000:
                        print("\n ======================================= \n")
                        print(str(a)+"\t"+str(b)+"\t"+str(c))
                        print(str(a*b*c))
                        print("\n ======================================= \n")
                        return a*b*c

main()

File: 9_special_pythagorean_triplet/9_special_pythagorean_triplet.c
// Problem is here: https://projecteuler.net/problem=9
// finds the product of a pythagorean triplet where its sum is 1000

#include <stdio.h>


int main()
{
  // a^2 + b^2 = c^2
  int a;
  int b;
  int c;

  for(a=1;a<333;a++) 
  {

    for (b=a;b<500;b++)
    {

      for (c=b;c<1000;c++)
      {
        long int a_squared = a*a;
        long int b_squared = b*b;
        long int c_squared = c*c;
        if ((a_squared+b_squared==c_squared) && (a+b+c==1000))
        {
          long int answer = a*b*c;
          printf("%ld\n",answer);
        }
      }
    }
  }
}
